app/routes/profile/$username/index.tsx

```diff
@@ -145,15 +118,42 @@ export const loader: LoaderFunction = async (
     "avatar",
     "background",
     "memberOf",
-    "teamMemberOfEvents",
     ...profile.publicFields,
   ];
 
-  let data: Partial<Profile> = {};
+  const eventRelationKeys = [
+    "teamMemberOfEvents",
+    "participatedEvents",
+    "contributedEvents",
+    "waitingForEvents",
+  ];
+
+  // TODO: Type issues, rework public fields
+  let data: Partial<
+    NonNullable<Awaited<ReturnType<typeof getProfileByUsername>>>
+  > = {};
   for (const key in profile) {
+    // Only show public fields if user is anon, show all fields if user is not anon
     if (mode !== "anon" || publicFields.includes(key)) {
-      // @ts-ignore <-- Partials allow undefined, Profile not
-      data[key] = profile[key];
+      // Event relations must be transformed
```

Comment:
use functions instead of comments (precise naming)

Reply:
Created a ticket:
see #379

Reply:
Resolved in 303171c